shared/bsp/bsp_pwm.cc

```diff
@@ -59,15 +59,15 @@ void PWM::Stop() {
 }
 
 void PWM::SetFrequency(uint32_t output_freq) {
-  this->output_freq_ = output_freq;
+  output_freq_ = output_freq;
```

Comment:
I think the `this` here add readability because of the level of similarities between the parameter and the class property.

Reply:
Em...Maybe? But since this is unnecessary, I still suggest to remove it.

Reply:
So if both the class property and the argument is called `output_freq`, then the `this` will be necessary. I still think this adds readability to indicate updating class properties.